python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9107,3 +9107,83 @@ def test_dataframe_pct_change(data, periods, fill_method):
     expected = pdf.pct_change(periods=periods, fill_method=fill_method)
 
     assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize(
+    ("key, value"),
+    [
+        (([0], ["x", "y"]), [10, 20],),
+        (([0, 2], ["x", "y"]), [[10, 30], [20, 40]],),
+    ],
+)
+def test_dataframe_loc_inplace_update(key, value):
+    gdf = cudf.DataFrame({"x": [1, 2, 3], "y": [4, 5, 6]})
+    pdf = gdf.to_pandas()
+
+    actual = gdf.loc[key] = value
+    expected = pdf.loc[key] = value
+
+    assert_eq(expected, actual)
```

Comment:
The usage of `loc` mainly focuses on that you can specify the rows by index labels. Here you are using the row number as integer labels, it works because gdf is just a `RangeIndex`. But I expect things might fail if you specify a non trivial index to it, like `index=list('abc')`, and specify to assign to the rows of `a` and `c`. Can you add a test to that?

Reply:
```
In [3]: gdf1 = cudf.DataFrame({"x" : [1,2,3], "y" : [4,5,6]}, index=list("abc"))
   ...: pdf1 =  gdf1.to_pandas()
   ...: gdf1
Out[3]: 
   x  y
a  1  4
b  2  5
c  3  6

In [4]: pdf1.loc[(["a"], ["x", "y"])] = [10, 20]
   ...: pdf1
Out[4]: 
    x   y
a  10  20
b   2   5
c   3   6

In [5]: gdf1.loc[(["a"], ["x", "y"]) ] = [10, 20]
   ...: gdf1
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [5], in <cell line: 1>()
----> 1 gdf1.loc[(["a"], ["x", "y"]) ] = [10, 20]
      2 gdf1

File ~/cudf/python/cudf/cudf/core/dataframe.py:133, in _DataFrameIndexer.__setitem__(self, key, value)
    131 if not isinstance(key, tuple):
    132     key = (key, slice(None))
--> 133 return self._setitem_tuple_arg(key, value)

File ~/compose/etc/conda/cuda_11.5/envs/rapids/lib/python3.8/contextlib.py:75, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     72 @wraps(func)
     73 def inner(*args, **kwds):
     74     with self._recreate_cm():
---> 75         return func(*args, **kwds)

File ~/cudf/python/cudf/cudf/core/dataframe.py:400, in _DataFrameLocIndexer._setitem_tuple_arg(self, key, value)
    389     raise ValueError(
    390         Template(
    391             "shape mismatch: value array of shape $value1 could \
   (...)
    397         )
    398     )
    399 for i, col in enumerate(columns_df._column_names):
--> 400     self._frame._data[col][key[0]] = value[:, i]

File ~/cudf/python/cudf/cudf/core/column/column.py:517, in ColumnBase.__setitem__(self, key, value)
    515     key = as_column(key)
    516     if not isinstance(key, cudf.core.column.NumericalColumn):
--> 517         raise ValueError(f"Invalid scatter map type {key.dtype}.")
    518     out = self._scatter_by_column(key, value_normalized)
    520 if out:

ValueError: Invalid scatter map type object.
```

^^ Is this expected behavior from our end or should we match pandas? @isVoid

Reply:
We should match pandas here. To me this seems like a common enough use case.